File: HomeWorks/src/Lesson6/Robot.java
package Lesson6;

import Lesson6.Hands.IHand;
import Lesson6.Heads.IHead;
import Lesson6.Legs.ILeg;

public class Robot implements IRobot {
    private IHead head;
    private IHand hand;
    private ILeg leg;
    private String name;

    public Robot(IHead head, IHand hand, ILeg leg, String name) {
        this.head = head;
        this.hand = hand;
        this.leg = leg;
        this.name = name;
    }

    public Robot(){}

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public IHead getHead() {
        return head;
    }

    public void setHead(IHead head) {
        this.head = head;
    }

    public IHand getHand() {
        return hand;
    }

    public void setHand(IHand hand) {
        this.hand = hand;
    }

    public ILeg getLeg() {
        return leg;
    }

    public void setLeg(ILeg leg) {
        this.leg = leg;
    }

    @Override
    public void action() {
        head.speak();
        hand.upHand();
        leg.step();
    }

    /**
    * Get the cost of a robot
    * @return int
    */
    @Override
    public int getPrice() {
        int price = head.getPrice() + hand.getPrice() + leg.getPrice();
        return price;
    }

}

File: HomeWorkKotlin/src/Robot/Robot.kt
package Robot

import Robot.Hands.IHand
import Robot.Heads.IHead
import Robot.Legs.ILeg

class Robot() : IRobot {
    private lateinit var head : IHead
    private lateinit var hand : IHand
    private lateinit var leg : ILeg
    private lateinit var name : String

    constructor(head_param : IHead, hand_param : IHand, leg_param : ILeg, name_param : String) : this() {
        head = head_param
        hand = hand_param
        leg = leg_param
        name = name_param

    }

    override fun action() {
        println("Меня зовут $name")
        head.speak()
        hand.upHand()
        leg.step()
    }

    fun getRobotName(): String = name

    override fun getRobotPrice(): Int = head.getHeadPrice() + hand.getHandPrice() + leg.getLegPrice()

}